docs/DeveloperGuide.md

```diff
@@ -1,29 +1,173 @@
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### Ui
+The `Ui` class handles user input and basic output messages. It can
+* Read user input
+* Output various messages to the user
 
-## Design & implementation
+![img.png](imgs/UiClass.png)
+
+### Storage Class
+![img.png](imgs/StorageClass.png)
+
+### Media Classes
+![img.png](imgs/MovieClass.png)
+
+### Commands
+The command component enables users to make changes to their review list. The command word is taken from the first word
+of the user input, and is processed through the `Parser` class. The class diagram shows how the Commands parent class is
+implemented, as well as its extended classes.
+![img.png](imgs/CommandsClass.png)
+
+#### Add
+The `add` command enables users to create new movie or TV Show reviews and add them to their review list. The following 
+is how the `add` command works:
+1. The `Parser` class takes in the user input, and parses this by using its `processUserInput` function.
+2. If the parsed command word is recognised to be the `add` command, the `executeAdd` function is called.
+3. Using the parsed command, `Parser` then determines whether the review to be added is a movie or Tv show. Otherwise, a 
+DukeException error is thrown.
+4. A new variable of type Media parent class, `toAdd`, is then created. The `AddCommand` function is then called to add
+the Movie or Tv Show `toAdd` into the review list, `reviewList`.
+![img_1.png](imgs/AddCommandSequence.png)
+
+#### Find
+The find command enables users to search for existing reviews that contains a given keyword. This is done through the
+following steps: 
+1. The `Parser` class takes in the user input, and parses this by using its `processUserInput` function.
+2. If the parsed command word is recognised to be the `add` command, the `executeFind` function is called and extracts
+the keyword to search for from the user input.
+3. `Parser` then creates an `executor` object by creating a new `FindCommand`. `Parser` then calls the `execute` 
+function from `executor` which loops through all reviews in the list and displays those containing the given keyword
+using the `toString()` function found in both the `Movie` and `TvShow` subclasses.
+![img.png](imgs/FindCommandSequence.png)
+
+#### List
+The list command outputs all inputted reviews per category:
+
+1. The list command sequence begins with a `Parser` object.
+2. The `Parser` object calls on `executeList` which is responsible for creating a `ListCommand` Object. It then calls on its `execute` method to execute the proper actions.
+3. In this `execute` function, it loops through all the stored reviews and separates each review based on category and formats using the proper `toString` method.
+4. Finally, an output string is returned and the `Parser` class calls on the `UI` object to print the output to the user.  
+![img_2.png](imgs/ListCommandSequence.png)
```

Comment:
This could be a bug, is this a self-invocation? 

![image](https://user-images.githubusercontent.com/53231968/198251339-f98e3342-aa2c-4192-bcd1-1ad9d5c20002.png)